<ui version="4.0" >
 <author></author>
 <comment></comment>
 <exportmacro></exportmacro>
 <class>gui</class>
 <widget class="QWidget" name="gui" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>296</width>
    <height>400</height>
   </rect>
  </property>
  <property name="sizePolicy" >
   <sizepolicy>
    <hsizetype>1</hsizetype>
    <vsizetype>1</vsizetype>
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize" >
   <size>
    <width>280</width>
    <height>400</height>
   </size>
  </property>
  <property name="windowTitle" >
   <string>ImageSpace</string>
  </property>
  <layout class="QVBoxLayout" >
   <property name="margin" >
    <number>8</number>
   </property>
   <property name="spacing" >
    <number>6</number>
   </property>
   <item>
    <widget class="QTabWidget" name="tabWidget" >
     <widget class="QWidget" name="tab" >
      <attribute name="title" >
       <string>Select</string>
      </attribute>
      <layout class="QGridLayout" >
       <property name="margin" >
        <number>8</number>
       </property>
       <property name="spacing" >
        <number>6</number>
       </property>
       <item row="0" column="0" >
        <widget class="QFrame" name="g1f" >
         <property name="minimumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="maximumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="frameShape" >
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="1" >
        <widget class="QFrame" name="g5f" >
         <property name="minimumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="maximumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="frameShape" >
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="1" >
        <widget class="QFrame" name="g2f" >
         <property name="minimumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="maximumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="frameShape" >
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
       <item row="3" column="1" >
        <widget class="QPushButton" name="renderButton" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>7</hsizetype>
           <vsizetype>7</vsizetype>
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize" >
          <size>
           <width>72</width>
           <height>25</height>
          </size>
         </property>
         <property name="toolTip" >
          <string>Start rendering</string>
         </property>
         <property name="text" >
          <string>Render</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0" >
        <widget class="QPushButton" name="loadButton" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>7</hsizetype>
           <vsizetype>7</vsizetype>
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize" >
          <size>
           <width>72</width>
           <height>25</height>
          </size>
         </property>
         <property name="toolTip" >
          <string>Load from a file</string>
         </property>
         <property name="text" >
          <string>Load</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2" >
        <widget class="QFrame" name="g3f" >
         <property name="minimumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="maximumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="frameShape" >
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="0" >
        <widget class="QFrame" name="g4f" >
         <property name="minimumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="maximumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="frameShape" >
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="2" >
        <widget class="QFrame" name="g6f" >
         <property name="minimumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="maximumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="frameShape" >
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="0" >
        <widget class="QFrame" name="g7f" >
         <property name="minimumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="maximumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="frameShape" >
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="1" >
        <widget class="QFrame" name="g8f" >
         <property name="minimumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="maximumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="frameShape" >
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="2" >
        <widget class="QFrame" name="g9f" >
         <property name="minimumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="maximumSize" >
          <size>
           <width>72</width>
           <height>72</height>
          </size>
         </property>
         <property name="frameShape" >
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
       <item row="3" column="2" >
        <widget class="QPushButton" name="saveButton" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>7</hsizetype>
           <vsizetype>7</vsizetype>
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize" >
          <size>
           <width>72</width>
           <height>25</height>
          </size>
         </property>
         <property name="toolTip" >
          <string>Save to a file</string>
         </property>
         <property name="text" >
          <string>Save</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab" >
      <attribute name="title" >
       <string>Render</string>
      </attribute>
      <layout class="QVBoxLayout" >
       <property name="margin" >
        <number>8</number>
       </property>
       <property name="spacing" >
        <number>6</number>
       </property>
       <item>
        <widget class="QGroupBox" name="groupBox" >
         <property name="title" >
          <string>Range</string>
         </property>
         <layout class="QGridLayout" >
          <property name="margin" >
           <number>8</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item row="1" column="1" >
           <widget class="QSpinBox" name="y1" >
            <property name="toolTip" >
             <string>Vertical maximum</string>
            </property>
            <property name="suffix" >
             <string/>
            </property>
            <property name="prefix" >
             <string>y1: </string>
            </property>
            <property name="maximum" >
             <number>999</number>
            </property>
            <property name="minimum" >
             <number>-998</number>
            </property>
            <property name="value" >
             <number>5</number>
            </property>
           </widget>
          </item>
          <item row="0" column="1" >
           <widget class="QSpinBox" name="x1" >
            <property name="toolTip" >
             <string>Horizontal maximum</string>
            </property>
            <property name="suffix" >
             <string/>
            </property>
            <property name="prefix" >
             <string>x1: </string>
            </property>
            <property name="maximum" >
             <number>999</number>
            </property>
            <property name="minimum" >
             <number>-998</number>
            </property>
            <property name="value" >
             <number>5</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0" >
           <widget class="QSpinBox" name="y0" >
            <property name="toolTip" >
             <string>Vertical minimum</string>
            </property>
            <property name="suffix" >
             <string/>
            </property>
            <property name="prefix" >
             <string>y0: </string>
            </property>
            <property name="maximum" >
             <number>998</number>
            </property>
            <property name="minimum" >
             <number>-999</number>
            </property>
            <property name="value" >
             <number>-5</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0" >
           <widget class="QSpinBox" name="x0" >
            <property name="toolTip" >
             <string>Horizontal minimum</string>
            </property>
            <property name="suffix" >
             <string/>
            </property>
            <property name="prefix" >
             <string>x0: </string>
            </property>
            <property name="maximum" >
             <number>998</number>
            </property>
            <property name="minimum" >
             <number>-999</number>
            </property>
            <property name="value" >
             <number>-5</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2" >
         <property name="title" >
          <string>Resolution</string>
         </property>
         <layout class="QGridLayout" >
          <property name="margin" >
           <number>8</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item row="0" column="1" >
           <widget class="QSpinBox" name="vres" >
            <property name="toolTip" >
             <string>Vertical</string>
            </property>
            <property name="suffix" >
             <string/>
            </property>
            <property name="prefix" >
             <string>V: </string>
            </property>
            <property name="maximum" >
             <number>4096</number>
            </property>
            <property name="minimum" >
             <number>64</number>
            </property>
            <property name="value" >
             <number>256</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0" >
           <widget class="QSpinBox" name="hres" >
            <property name="toolTip" >
             <string>Horizontal</string>
            </property>
            <property name="suffix" >
             <string/>
            </property>
            <property name="prefix" >
             <string>H: </string>
            </property>
            <property name="maximum" >
             <number>4096</number>
            </property>
            <property name="minimum" >
             <number>64</number>
            </property>
            <property name="value" >
             <number>256</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="alpha" >
         <property name="toolTip" >
          <string>Whether the alpha channel should be used or not</string>
         </property>
         <property name="text" >
          <string>Alpha channel</string>
         </property>
         <property name="checked" >
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer>
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" >
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab" >
      <attribute name="title" >
       <string>Math</string>
      </attribute>
      <layout class="QVBoxLayout" >
       <property name="margin" >
        <number>8</number>
       </property>
       <property name="spacing" >
        <number>6</number>
       </property>
       <item>
        <widget class="QTabWidget" name="tabWidget_2" >
         <widget class="QWidget" name="tab" >
          <attribute name="title" >
           <string>Squashing</string>
          </attribute>
          <layout class="QVBoxLayout" >
           <property name="margin" >
            <number>8</number>
           </property>
           <property name="spacing" >
            <number>6</number>
           </property>
           <item>
            <widget class="QCheckBox" name="op_tanh" >
             <property name="text" >
              <string>Tanh</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="op_sin" >
             <property name="text" >
              <string>Sin</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="op_cos" >
             <property name="text" >
              <string>Cos</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="op_gauss" >
             <property name="text" >
              <string>Gaussian</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer>
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" >
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tab" >
          <attribute name="title" >
           <string>Operations</string>
          </attribute>
          <layout class="QVBoxLayout" >
           <property name="margin" >
            <number>8</number>
           </property>
           <property name="spacing" >
            <number>6</number>
           </property>
           <item>
            <widget class="QCheckBox" name="op_sum" >
             <property name="text" >
              <string>Sum</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="op_prod" >
             <property name="text" >
              <string>Product</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="op_qprod" >
             <property name="text" >
              <string>Quaternion Product</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="op_sh0" >
             <property name="text" >
              <string>Shift Red</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="op_sh1" >
             <property name="text" >
              <string>Shift Green</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="op_sh2" >
             <property name="text" >
              <string>Shift Blue</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="op_sh3" >
             <property name="text" >
              <string>Shift Alpha</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer>
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" >
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tab" >
          <attribute name="title" >
           <string>Permutations</string>
          </attribute>
          <layout class="QVBoxLayout" >
           <property name="margin" >
            <number>8</number>
           </property>
           <property name="spacing" >
            <number>6</number>
           </property>
           <item>
            <widget class="QCheckBox" name="op_perm01" >
             <property name="text" >
              <string>Permutate Red/Green</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="op_perm02" >
             <property name="text" >
              <string>Permutate Red/Blue</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="op_perm03" >
             <property name="text" >
              <string>Permutate Red/Alpha</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="op_perm12" >
             <property name="text" >
              <string>Permutate Green/Blue</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="op_perm13" >
             <property name="text" >
              <string>Permutate Green/Alpha</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="op_perm23" >
             <property name="text" >
              <string>Permutate Blue/Alpha</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer>
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" >
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab" >
      <attribute name="title" >
       <string>Options</string>
      </attribute>
      <layout class="QVBoxLayout" >
       <property name="margin" >
        <number>8</number>
       </property>
       <property name="spacing" >
        <number>6</number>
       </property>
       <item>
        <widget class="QGroupBox" name="groupBox_3" >
         <property name="title" >
          <string>Subnodes</string>
         </property>
         <layout class="QVBoxLayout" >
          <property name="margin" >
           <number>8</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item>
           <widget class="QSpinBox" name="decay" >
            <property name="toolTip" >
             <string>Low values increase the syntax tree depth</string>
            </property>
            <property name="suffix" >
             <string>%</string>
            </property>
            <property name="prefix" >
             <string>Subnode Prob. Decay: </string>
            </property>
            <property name="maximum" >
             <number>99</number>
            </property>
            <property name="minimum" >
             <number>20</number>
            </property>
            <property name="value" >
             <number>20</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="maxSubnodes" >
            <property name="toolTip" >
             <string>Higher values increase the image complexity</string>
            </property>
            <property name="suffix" >
             <string/>
            </property>
            <property name="prefix" >
             <string>Max. #of Subnodes: </string>
            </property>
            <property name="maximum" >
             <number>50</number>
            </property>
            <property name="minimum" >
             <number>2</number>
            </property>
            <property name="value" >
             <number>3</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_3_2" >
         <property name="title" >
          <string>Mutation Probability</string>
         </property>
         <layout class="QVBoxLayout" >
          <property name="margin" >
           <number>8</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item>
           <widget class="QSpinBox" name="mutation" >
            <property name="toolTip" >
             <string>The probability with which a node is mutated</string>
            </property>
            <property name="suffix" >
             <string>%</string>
            </property>
            <property name="prefix" >
             <string/>
            </property>
            <property name="maximum" >
             <number>100</number>
            </property>
            <property name="minimum" >
             <number>1</number>
            </property>
            <property name="value" >
             <number>30</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="autoOpen" >
         <property name="toolTip" >
          <string>Whether to open new renderers or not</string>
         </property>
         <property name="text" >
          <string>Open new renderers</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="textLabel1" >
         <property name="text" >
          <string>Homepage:
imagespace.sourceforge.net
info@subsymbolics.com</string>
         </property>
         <property name="alignment" >
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <pixmapfunction></pixmapfunction>
 <resources/>
 <connections/>
</ui>
